kafka/src/main/java/io/micronaut/configuration/kafka/admin/KafkaNewTopics.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.configuration.kafka.admin;
+
+import io.micronaut.context.ApplicationContext;
+import io.micronaut.context.annotation.Context;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.annotation.Experimental;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.CreateTopicsOptions;
+import org.apache.kafka.clients.admin.CreateTopicsResult;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.function.Predicate.not;
+
+/**
+ * Creates Kafka topics via {@link AdminClient}.
+ *
+ * @author Guillermo Calvo
+ * @since 5.2
+ */
+@Context
+@Requires(bean = AdminClient.class)
+public class KafkaNewTopics {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KafkaNewTopics.class);
+
+    @Nullable
+    private final CreateTopicsOptions options;
+
+    @Nullable
+    private final CreateTopicsResult result;
+
+    /**
+     * @param context The application context.
+     * @param options Optional {@link CreateTopicsOptions}.
+     * @param topics  Optional list of {@link NewTopic} beans.
+     */
+    KafkaNewTopics(
+        @NonNull ApplicationContext context,
```

Comment:
Please, inject the `AdminClient` instead of the `ApplicationContext`.

Reply:
That's exactly what I wanted to avoid. You see, `KafkaNewsTopics` is a `@Context` bean. But I don't want to instantiate a new `AdminClient` unless it's an absolute need (if there are some `NewTopic` beans in the application context).

The reason is `AdminClient` connects to the Kafka broker upon creation and needs admin permissions to succeed. Doing what you ask here would make all applications create an admin connection at startup. We recently updated our health indicator to prevent that from happening.

Reply:
Why not annotating with `@Requires(bean = NewTopic.class`?

Reply:
I didn't do that because I wanted users to be able to `@Inject` this guy regardless of new topics being created -- I was thinking maybe `NewTopic` beans could be created for `@EachProperty`, and in some envs there would be none.

But I guess they can just `@Inject` `@Nullable`, so I'm going to try that 👍